compat/src/suspense.js

```diff
@@ -43,12 +43,11 @@ options.unmount = function(vnode) {
 
 function detachedClone(vnode, detachedParent, parentDom) {
 	if (vnode) {
-		if (vnode._component && vnode._component.__hooks) {
-			vnode._component.__hooks._list.forEach(effect => {
-				if (typeof effect._cleanup == 'function') effect._cleanup();
-			});
-
+		if (vnode._component && vnode._component.__hooks && !vnode._dom) {
```

Comment:
This means that we reset hooks-state only on first-mount, not really sure if this is the most ideal way to detect this but for now we do it like this.

So during the traversal up during initial mount we reset all hooks

Reply:
Ah dang yea. In restructure we could some of our flags to better detect this :/